pkg/agent/agent.go

```diff
@@ -610,11 +610,11 @@ func (i *Initializer) configureGatewayInterface(gatewayIface *interfacestore.Int
 	gatewayIface.IPs = []net.IP{}
 	if i.networkConfig.TrafficEncapMode.IsNetworkPolicyOnly() {
 		// Assign IP to gw as required by SpoofGuard.
-		if i.nodeConfig.NodeIPv4Addr != nil {
+		if i.nodeConfig.NodeTransportIPv4Addr != nil {
```

Comment:
Is it a bug?

Reply:
This PR is to expected resolve a canddate issue if the transport interface different from the management interface.

About this line, I change it to `NodeTransportIPv4Addr` because the next lines (https://github.com/antrea-io/antrea/pull/3000/files/02a6009e067ca734713f6675a78739ea487c7fc4#diff-18bf746206c8ac217eb16ffb0cf91a6e676e05a7b517dc70aef0260d0871028fL614) actually uses `NodeTransportIPv4Addr` instead of `NodeIPv4Addr`

Reply:
I meant this line specifically. It looks like a bug, but I got now NodeTransportIPv4Addr should be at least set to NodeIPv4Addr already.

Reply:
Yes, the `NodeTransportIPv4Addr` is set with "NodeIPv4Addr" if no transport interface is configured, then we could use `NodeTransportIPv4Addr` in all configurations whch is focusing on transporting the traffic.